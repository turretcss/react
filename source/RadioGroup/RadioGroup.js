import React, { Component } from 'react'

import { Radio } from '../Control'

let refCount = 0

class RadioGroup extends Component {
  constructor() {
    super()
    this._name = `_radio-group-${++refCount}_`
  }

  render() {
    const { options = [], name, value, onChange, ...props } = this.props
    return options.map(option => (
      <Radio
        {...props}
        {...option}
        name={name || this._name}
        key={option.value}
        value={value == option.value}
        onChange={() => onChange(option.value)}
        label={option.label}
      />
    ))
  }
}

export default RadioGroup
