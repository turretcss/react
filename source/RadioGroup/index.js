export { default } from './RadioGroup'
export { default as RadioGroup } from './RadioGroup'
