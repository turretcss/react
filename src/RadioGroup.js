import React, { Component } from 'react'

import { Radio } from './Control'

let refCount = 0

class RadioGroup extends Component {
  constructor() {
    super()
    this._name = `_radio-group-${++refCount}_`
  }

  render() {
    const { options = [], value, onChange, ...props } = this.props
    return options.map(option => (
      <Radio
        {...props}
        name={this._name}
        key={option.value}
        value={value == option.value}
        onChange={() => onChange(option.value)}
        text={option.text}
      />
    ))
  }
}

export default RadioGroup
