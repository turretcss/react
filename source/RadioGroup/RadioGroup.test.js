import React from 'react'
import renderer from 'react-test-renderer'
import { renderToStaticMarkup } from 'react-dom/server'

import RadioGroup from './RadioGroup'

const options = [
  {
    value: null,
    label: 'Auto',
  },
  {
    value: 's',
    label: 'Small',
  },
  {
    value: 'l',
    label: 'Large',
  },
]

describe('RadioGroup', () => {
  it('renders nothing if no options supplied', () => {
    const result = renderToStaticMarkup(<RadioGroup />)
    const expected = ''
    expect(result).toEqual(expected)
  })

  it('renders array of radios', () => {
    const tree = renderer.create(<RadioGroup options={options} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders array of radios with "Small" option selected', () => {
    const tree = renderer.create(<RadioGroup value="s" options={options} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders custom name prop', () => {
    const tree = renderer.create(<RadioGroup name="custom-name" options={options} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('passes unknown props down to radios', () => {
    const tree = renderer.create(<RadioGroup options={options} size="xs" disabled color="error" />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('calls onChange when input clicked', () => {
    let newValue = null
    const testInstance = renderer.create(<RadioGroup onChange={value => (newValue = value)} options={options} />).root
    // click radio with value "s"
    testInstance.findAllByType('input')[1].props.onClick({ target: { checked: true } })
    expect(newValue).toEqual('s')
  })
})
