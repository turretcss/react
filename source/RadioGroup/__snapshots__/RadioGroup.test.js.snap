// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioGroup passes unknown props down to radios 1`] = `
Array [
  <label
    className="control radio control-xs error"
  >
    <input
      checked={true}
      disabled={true}
      name="_radio-group-5_"
      onChange={[Function]}
      onClick={[Function]}
      type="radio"
    />
    <span
      className="control-indicator"
    />
    <span
      className="control-label"
    >
      Auto
    </span>
  </label>,
  <label
    className="control radio control-xs error"
  >
    <input
      checked={false}
      disabled={true}
      name="_radio-group-5_"
      onChange={[Function]}
      onClick={[Function]}
      type="radio"
    />
    <span
      className="control-indicator"
    />
    <span
      className="control-label"
    >
      Small
    </span>
  </label>,
  <label
    className="control radio control-xs error"
  >
    <input
      checked={false}
      disabled={true}
      name="_radio-group-5_"
      onChange={[Function]}
      onClick={[Function]}
      type="radio"
    />
    <span
      className="control-indicator"
    />
    <span
      className="control-label"
    >
      Large
    </span>
  </label>,
]
`;

exports[`RadioGroup renders array of radios 1`] = `
Array [
  <label
    className="control radio"
  >
    <input
      checked={true}
      disabled={undefined}
      name="_radio-group-2_"
      onChange={[Function]}
      onClick={[Function]}
      type="radio"
    />
    <span
      className="control-indicator"
    />
    <span
      className="control-label"
    >
      Auto
    </span>
  </label>,
  <label
    className="control radio"
  >
    <input
      checked={false}
      disabled={undefined}
      name="_radio-group-2_"
      onChange={[Function]}
      onClick={[Function]}
      type="radio"
    />
    <span
      className="control-indicator"
    />
    <span
      className="control-label"
    >
      Small
    </span>
  </label>,
  <label
    className="control radio"
  >
    <input
      checked={false}
      disabled={undefined}
      name="_radio-group-2_"
      onChange={[Function]}
      onClick={[Function]}
      type="radio"
    />
    <span
      className="control-indicator"
    />
    <span
      className="control-label"
    >
      Large
    </span>
  </label>,
]
`;

exports[`RadioGroup renders array of radios with "Small" option selected 1`] = `
Array [
  <label
    className="control radio"
  >
    <input
      checked={false}
      disabled={undefined}
      name="_radio-group-3_"
      onChange={[Function]}
      onClick={[Function]}
      type="radio"
    />
    <span
      className="control-indicator"
    />
    <span
      className="control-label"
    >
      Auto
    </span>
  </label>,
  <label
    className="control radio"
  >
    <input
      checked={true}
      disabled={undefined}
      name="_radio-group-3_"
      onChange={[Function]}
      onClick={[Function]}
      type="radio"
    />
    <span
      className="control-indicator"
    />
    <span
      className="control-label"
    >
      Small
    </span>
  </label>,
  <label
    className="control radio"
  >
    <input
      checked={false}
      disabled={undefined}
      name="_radio-group-3_"
      onChange={[Function]}
      onClick={[Function]}
      type="radio"
    />
    <span
      className="control-indicator"
    />
    <span
      className="control-label"
    >
      Large
    </span>
  </label>,
]
`;

exports[`RadioGroup renders custom name prop 1`] = `
Array [
  <label
    className="control radio"
  >
    <input
      checked={true}
      disabled={undefined}
      name="custom-name"
      onChange={[Function]}
      onClick={[Function]}
      type="radio"
    />
    <span
      className="control-indicator"
    />
    <span
      className="control-label"
    >
      Auto
    </span>
  </label>,
  <label
    className="control radio"
  >
    <input
      checked={false}
      disabled={undefined}
      name="custom-name"
      onChange={[Function]}
      onClick={[Function]}
      type="radio"
    />
    <span
      className="control-indicator"
    />
    <span
      className="control-label"
    >
      Small
    </span>
  </label>,
  <label
    className="control radio"
  >
    <input
      checked={false}
      disabled={undefined}
      name="custom-name"
      onChange={[Function]}
      onClick={[Function]}
      type="radio"
    />
    <span
      className="control-indicator"
    />
    <span
      className="control-label"
    >
      Large
    </span>
  </label>,
]
`;
